src/jdk.compiler/share/classes/com/sun/tools/javac/code/Symtab.java

```diff
@@ -274,7 +271,13 @@ public VarSymbol getClassField(Type type, Types types) {
                        I1, I2, ... In, V.class is typed to be Class<? extends Object & I1 & I2 .. & In>
                     */
                     if (type.isValue()) {
-                        Type it = types.makeIntersectionType(((ClassType)type).interfaces_field, true);
+                        ClassType ct = (ClassType) type;
+                        Type it;
+                        if (ct.interfaces_field == null || ct.interfaces_field.isEmpty()) {
```

Comment:
I'm assuming that you are using this to indicate no top interfaces. Is this safe (non-hackish)?

Reply:
Yes, it is safe. Basically earlier on all value classes had at least one interface (InlineObject) and possibly more. Now the magic top type InlineObject going away, values may have no super interfaces at all and this would upset makeIntersectionType which first asserts that it has not been called with an empty list of types.